sdk/storage/azblob/zc_list_blobs_wrapper.go

```diff
@@ -68,19 +84,32 @@ func (p listBlobsHierarchySegmentAutoPager) Go() {
 			for _, v := range *resp.EnumerationResults.Segment.BlobItems {
```

Comment:
Hey @adreed-msft ,

The response of pacer we've written in the Listing API(Hierarchical) is a channel of `BlobItemInternal` struct. 

Actual response looks like this. 
```
type ListBlobsHierarchySegmentResponse struct {
   ContainerName   *string                   `xml:"ContainerName,attr"`
   Delimiter       *string                   `xml:"Delimiter"`
   Marker          *string                   `xml:"Marker"`
   MaxResults      *int32                    `xml:"MaxResults"`
   NextMarker      *string                   `xml:"NextMarker"`
   Prefix          *string                   `xml:"Prefix"`
   Segment         *BlobHierarchyListSegment `xml:"Blobs"`
   ServiceEndpoint *string                   `xml:"ServiceEndpoint,attr"`
}
```
```
type BlobHierarchyListSegment struct {
   BlobItems    *[]BlobItemInternal `xml:"Blob"`
   BlobPrefixes *[]BlobPrefix       `xml:"BlobPrefix"`
}
```
I see that we're only iterating over BlobItems so we don't have other attributes in the response. This is problematic as we're returning only a fragment of actual response. Many of our use cases involve other attributes as well.